### PR TITLE
Disable scale-application, add-unit, remove-unit for daemonset;

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - ~[ ] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?~
 - ~[ ] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?~
 - ~[ ] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?~
 - [ x] Do comments answer the question of why design decisions were made?

----

## Description of change

Disable scale-application, add-unit, remove-unit for daemonset;

## QA steps

```console
$ yml2json /tmp/charm-builds/mariadb-k8s/metadata.yaml --pretty | jq '.deployment'
{
  "type": "daemon",
  "min-version": "1.10.1",
  "service": "omit"
}

$ juju add-model t1 microk8s && juju deploy /tmp/charm-builds/mariadb-k8s/ --debug  --resource mysql_image=mariadb

$ juju add-unit mariadb-k8s -n 3
ERROR can not add unit: scale a "daemon" charm not supported

$ juju remove-unit mariadb-k8s --num-units 3
ERROR can not remove unit: scale a "daemon" charm not supported

$ juju scale-application mariadb-k8s 3 -m t1
ERROR could not scale application "mariadb-k8s": scale a "daemon" charm not supported

```

## Documentation changes

Yes

## Bug reference

None

### CODE REVIEW

File: apiserver/facades/client/application/application.go [hunk 1]
```diff
@@ -1746,8 +1746,13 @@ func (api *APIBase) ScaleApplications(args params.ScaleApplicationsParams) (para
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		if ch.Meta().Deployment != nil && ch.Meta().Deployment.DeploymentMode == charm.ModeOperator {
-			return nil, errors.New("cannot scale an operator charm")
+		if ch.Meta().Deployment != nil {
+			if ch.Meta().Deployment.DeploymentMode == charm.ModeOperator {
+				return nil, errors.NotSupportedf("scale an %q charm", charm.ModeOperator)
```
Comment (wallyworld) [inline apiserver/facades/client/application/application.go:1751]: "scaling an %q application"

File: apiserver/facades/client/application/application.go [hunk 2]
```diff
@@ -1746,8 +1746,13 @@ func (api *APIBase) ScaleApplications(args params.ScaleApplicationsParams) (para
...
+			}
+			if ch.Meta().Deployment.DeploymentType == charm.DeploymentDaemon {
+				return nil, errors.NotSupportedf("scale a %q charm", charm.DeploymentDaemon)
```
Comment (wallyworld) [inline apiserver/facades/client/application/application.go:1754]: "scaling a %q appication"

**Comment by ycliuhw** [comment]
[disable juju scale and add[remove]-unit for daemonsets and operator charm](https://trello.com/c/N4B9q7PT/1051-disable-juju-scale-and-addremove-unit-for-daemonsets-and-operator-charm)

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**